## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 439883`
Commit: [`2853952480dc2dfd745de4fef8896a0704f53791`](https://github.com/NixOS/nixpkgs/commit/2853952480dc2dfd745de4fef8896a0704f53791) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/2853952480dc2dfd745de4fef8896a0704f53791..pull/439883/head))
Merge: [`7f6953dd27b70abfd878d1d17f949877ba250f6a`](https://github.com/NixOS/nixpkgs/commit/7f6953dd27b70abfd878d1d17f949877ba250f6a)

Logs: https://github.com/brianmcgillion/nixpkgs-review-gha/actions/runs/17437759755


---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>cfclient</li>
    <li>cfclient.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.cflib</li>
    <li>python312Packages.cflib.dist</li>
    <li>python313Packages.cflib</li>
    <li>python313Packages.cflib.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>cfclient</summary>
<pre>adding &#x27;cfzmq/__init__.py&#x27;
adding &#x27;cfzmq/__main__.py&#x27;
adding &#x27;cfclient-2025.2.dist-info/METADATA&#x27;
adding &#x27;cfclient-2025.2.dist-info/WHEEL&#x27;
adding &#x27;cfclient-2025.2.dist-info/entry_points.txt&#x27;
adding &#x27;cfclient-2025.2.dist-info/top_level.txt&#x27;
adding &#x27;cfclient-2025.2.dist-info/RECORD&#x27;
removing build/bdist.linux-x86_64/wheel
Successfully built cfclient-2025.2-py3-none-any.whl
Finished creating a wheel...
/build/source/dist /build/source
Unpacking to: unpacked/cfclient-2025.2...OK
Repacking wheel as ./cfclient-2025.2-py3-none-any.whl...OK
/build/source
Finished executing pypaBuildPhase
Running phase: pythonRuntimeDepsCheckHook
@nix { &quot;action&quot;: &quot;setPhase&quot;, &quot;phase&quot;: &quot;pythonRuntimeDepsCheckHook&quot; }
Executing pythonRuntimeDepsCheck
Checking runtime dependencies for cfclient-2025.2-py3-none-any.whl
  - pyzmq~=26.0 not satisfied by version 27.0.1</pre>
</details>
</details>


---
### `aarch64-linux`
<details>
  <summary>:x: 6 packages failed to build:</summary>
  <ul>
    <li>cfclient</li>
    <li>cfclient.dist</li>
    <li>python312Packages.cflib</li>
    <li>python312Packages.cflib.dist</li>
    <li>python313Packages.cflib</li>
    <li>python313Packages.cflib.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python312Packages.cflib</summary>
<pre>test/localization/lighthouse_test_base.py:47: in assertPosesAlmostEqual
    self.assertAlmostEqual(0.0, np.linalg.norm(translation_diff), places,
E   AssertionError: 0.0 != np.float64(nan) within 3 places (np.float64(nan) difference) : Translation different, expected: [ 1.  0. -1.], actual: [nan nan nan]
=============================== warnings summary ===============================
cflib/localization/lighthouse_bs_geo.py:28
  /build/source/cflib/localization/lighthouse_bs_geo.py:28: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools&lt;81.
    import pkg_resources

test/localization/lighthouse_test_base.py:26
  /build/source/test/localization/lighthouse_test_base.py:26: DeprecationWarning: Please import `Rotation` from the `scipy.spatial.transform` namespace; the `scipy.spatial.transform.rotation` namespace is deprecated and will be removed in SciPy 2.0.0.
    from scipy.spatial.transform.rotation import Rotation

test/localization/test_ippe_cf.py::TestIppeCf::test_that_pose_is_found
  /build/source/cflib/localization/_ippe.py:295: RuntimeWarning: invalid value encountered in sqrt
    b = np.vstack((np.sqrt(h[0, 0]), np.sqrt(h[1, 1])))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test/localization/test_ippe_cf.py::TestIppeCf::test_that_pose_is_found - AssertionError: 0.0 != np.float64(nan) within 3 places (np.float64(nan) dif...
================== 1 failed, 186 passed, 3 warnings in 6.25s ===================</pre>
</details>
<details>
<summary>python313Packages.cflib</summary>
<pre>test/localization/lighthouse_test_base.py:47: in assertPosesAlmostEqual
    self.assertAlmostEqual(0.0, np.linalg.norm(translation_diff), places,
E   AssertionError: 0.0 != np.float64(nan) within 3 places (np.float64(nan) difference) : Translation different, expected: [ 1.  0. -1.], actual: [nan nan nan]
=============================== warnings summary ===============================
cflib/localization/lighthouse_bs_geo.py:28
  /build/source/cflib/localization/lighthouse_bs_geo.py:28: UserWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html. The pkg_resources package is slated for removal as early as 2025-11-30. Refrain from using this package or pin to Setuptools&lt;81.
    import pkg_resources

test/localization/lighthouse_test_base.py:26
  /build/source/test/localization/lighthouse_test_base.py:26: DeprecationWarning: Please import `Rotation` from the `scipy.spatial.transform` namespace; the `scipy.spatial.transform.rotation` namespace is deprecated and will be removed in SciPy 2.0.0.
    from scipy.spatial.transform.rotation import Rotation

test/localization/test_ippe_cf.py::TestIppeCf::test_that_pose_is_found
  /build/source/cflib/localization/_ippe.py:295: RuntimeWarning: invalid value encountered in sqrt
    b = np.vstack((np.sqrt(h[0, 0]), np.sqrt(h[1, 1])))

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED test/localization/test_ippe_cf.py::TestIppeCf::test_that_pose_is_found - AssertionError: 0.0 != np.float64(nan) within 3 places (np.float64(nan) dif...
================== 1 failed, 186 passed, 3 warnings in 6.24s ===================</pre>
</details>
</details>

--- a/pkgs/by-name/cf/cfclient/package.nix
+++ b/pkgs/by-name/cf/cfclient/package.nix
@@ -37,6 +37,7 @@ python3Packages.buildPythonApplication rec {
   pythonRelaxDeps = [
     "numpy"
     "pyqt6"
+    "pyzmq"
     "vispy"
   ];
 

